app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -695,5 +696,27 @@ def set_vm_interfaces_attributes(vm, vm_attrs)
       end
       vm_attrs
     end
+
+    def set_vm_volumes_attributes(vm, vm_attrs)
+      template = template(vm_attrs[:template]) if vm_attrs[:template]
+      template_volumes_name_id = {}
+      template&.volumes&.each { |volume| template_volumes_name_id[volume.name] = volume.id }
+      volumes = vm.volumes || []
+      vm_attrs[:volumes_attributes] = volumes.each_with_index.each_with_object({}) do |(volume, index), hsh|
+        id = nil
+        id = template_volumes_name_id[volume.name] if template_volumes_name_id.key? volume.name
+        volume_attrs = {
```

Comment:
why not set ` hsh[index.to_s]` right away?

Reply:
we also have a very close structure being built in `normalize_vm_attrs`, maybe this can be reused if it's in a method?

Reply:
@orrabin     normalize_vm_attrs(vm_attrs) create a different volumes hash: 
 "volumes_attributes"=>
  {"1577176705633"=>
    {"size"=>"1073741824",
     "storage_domain_id"=>"f676fbc4-7351-4a79-9f8e-38cd0c67ed3f",
     "storage_domain_name"=>"vserver-spider-vol",
     "preallocate"=>false,
     "bootable"=>nil}}}


it's kind of similar but different: 

 :volumes_attributes=>
  {"0"=>
    {:size_gb=>1,
     :storage_domain=>"e257123b-b381-4326-b550-9c9af53a80d7",
     :preallocate=>"1",
     :wipe_after_delete=>"false",
     :interface=>"virtio_scsi",
     :bootable=>"true",
     :id=>nil}}}